packages/backend/src/auto-peering/service.ts

```diff
@@ -39,8 +56,110 @@ export async function createAutoPeeringService(
   }
 
   return {
-    acceptPeeringRequest: (args: PeeringRequestArgs) =>
-      acceptPeeringRequest(deps, args)
+    acceptPeeringRequest: (args) => acceptPeeringRequest(deps, args),
+    initiatePeeringRequest: (args) => initiatePeeringRequest(deps, args)
+  }
+}
+
+async function initiatePeeringRequest(
+  deps: ServiceDependencies,
+  args: InitiatePeeringRequestArgs
+): Promise<Peer | AutoPeeringError> {
+  const { peerUrl } = args
+
+  const asset = await deps.assetService.get(args.assetId)
+
+  if (!asset) {
+    return AutoPeeringError.UnknownAsset
+  }
+
+  const httpToken = uuid()
+
+  const peeringDetailsOrError = await sendPeeringRequest(deps, {
+    peerUrl,
+    ilpConnectorAddress: deps.config.ilpConnectorAddress,
+    staticIlpAddress: deps.config.ilpAddress,
+    asset: { code: asset.code, scale: asset.scale },
+    httpToken
+  })
+
+  if (isAutoPeeringError(peeringDetailsOrError)) {
+    return peeringDetailsOrError
+  }
+
+  const peerOrError = await deps.peerService.create({
+    maxPacketAmount: args.maxPacketAmount,
+    name: args.name,
+    assetId: asset.id,
+    staticIlpAddress: peeringDetailsOrError.staticIlpAddress,
+    http: {
+      incoming: {
+        authTokens: [httpToken]
+      },
+      outgoing: {
+        authToken: httpToken,
```

Comment:
I don't think these should be the same token

Reply:
At a high level, it doesn't make a difference, but now that I think about it, this matters in supporting the case where if the receiving peer (peer 2) has already registered the peer (peer 1) that is making the request: https://github.com/interledger/rafiki/blob/d5089e2a94c94c9c0815c1fd5b7a351bf833b309/packages/backend/src/auto-peering/service.ts#L194-L209

to support this functionality, we'd need to send back the outgoing token that peer 2 would use during outgoing packets. Going to update this :+1: